dts/bindings/gpio/gpio-controller.yaml

```
@@ -26,3 +26,17 @@ properties:
           "ngpios = <18>;" informs the driver that only the first 18 GPIOs, at
           local offset 0 .. 17, are in use.  For cases in which there might be
           holes in the slot range, this value should be the max slot number-1.
+    gpio-reserved-ranges:
+      type: array
+      required: false
+      description: |
+          If these GPIOs do not happen to be the first N GPIOs at offset
+          0...N-1, an additional set of tuples is needed to specify which GPIOs
+          are unusable. This property indicates the start and size of the GPIOs
+          that can't be used.
```

```suggestion
          If not all the GPIOs at offsets 0...N-1 are usable for ngpios = <N>, then
          this property contains an additional set of tuples which specify which GPIOs
          are unusable. This property indicates the start and size of the GPIOs
          that can't be used.

          For example, setting "gpio-reserved-ranges = <3 2>, <10 1>;" means that
          GPIO offsets 3, 4, and 10 are not usable, even if ngpios = <18>.
```

I think since zephyr binding syntax has properties with standalone descriptions, it's a bit easier to read when they're described in a more standalone way like this.